api/loader/fileloader_test.go

```diff
@@ -201,6 +201,30 @@ func TestLoaderBadRelative(t *testing.T) {
 	require.Error(err)
 }
 
+func TestLoaderAbsolute(t *testing.T) {
```

Comment:
Could this test include a case where an absolute path contains a part that's relative?
For example: `/foo/project/subdir2/subdir3/../subdir1`. Is this something that should succeed or fail for this implementation?

Reply:
I added two more tests:
- one where we escape the root using relative parts in the absolute path
- one where we stay in the root using relative parts in the absolute path

Let me know if this doesn't clarify your question or if further tests are needed.

Reply:
New tests make sense to me, thanks for updating!